src/lib/selector/fastAuthWalletConnection.ts

```diff
@@ -0,0 +1,335 @@
+import { KeyPair, transactions, utils } from 'near-api-js';
+const { SCHEMA } = transactions;
+import { ConnectedWalletAccount } from 'near-api-js';
+
+const LOGIN_WALLET_URL_SUFFIX = '/login/';
+const LOCAL_STORAGE_KEY_SUFFIX = '_wallet_auth_key';
+const PENDING_ACCESS_KEY_PREFIX = 'pending_key'; // browser storage key for a pending access key (i.e. key has been generated but we are not sure it was added yet)
+
+interface SignInOptions {
+  contractId?: string;
+  methodNames?: string[];
+  // TODO: Replace following with single callbackUrl
+  successUrl?: string;
+  failureUrl?: string;
+  email?: string;
+  accountId?: string;
+  isRecovery?: boolean;
+}
+
+/**
+ * Information to send NEAR wallet for signing transactions and redirecting the browser back to the calling application
+ */
+interface RequestSignTransactionsOptions {
+  /** list of transactions to sign */
+  transactions: transactions.Transaction[];
+  /** url NEAR Wallet will redirect to after transaction signing is complete */
+  callbackUrl?: string;
+  /** meta information NEAR Wallet will send back to the application. `meta` will be attached to the `callbackUrl` as a url search param */
+  meta?: string;
+}
+
+export class FastAuthWalletConnection {
+  /** @hidden */
+  _walletBaseUrl: string;
+
+  /** @hidden */
+  _authDataKey: string;
+
+  /** @hidden */
+  _keyStore: keyStores.KeyStore;
+
+  /** @hidden */
+  _authData: { accountId?: string; allKeys?: string[] };
+
+  /** @hidden */
+  _networkId: string;
+
+  /** @hidden */
+  // _near: Near;
+  _near: Near;
+
+  /** @hidden */
+  _connectedAccount: ConnectedWalletAccount;
+
+  /** @hidden */
+  _completeSignInPromise: Promise<void>;
+
+  /** @hidden */
+  _iframe: HTMLIFrameElement;
+
+  constructor(near: Near, appKeyPrefix: string) {
+    if (typeof appKeyPrefix !== 'string') {
+      throw new Error(
+        'Please define a clear appKeyPrefix for this WalletConnection instance as the second argument to the constructor',
+      );
+    }
+
+    if (typeof window === 'undefined') {
+      return new Proxy(this, {
+        get(target, property) {
+          if (property === 'isSignedIn') {
+            return () => false;
+          }
+          if (property === 'getAccountId') {
+            return () => '';
+          }
+          if (
+            target[property as keyof FastAuthWalletConnection] &&
+            typeof target[property as keyof FastAuthWalletConnection] === 'function'
+          ) {
+            return () => {
+              throw new Error(
+                'No window found in context, please ensure you are using WalletConnection on the browser',
+              );
+            };
+          }
+          return target[property as keyof FastAuthWalletConnection];
+        },
+      });
+    }
+    const iframe = document.createElement('iframe');
+    iframe.allow = 'publickey-credentials-create *; publickey-credentials-get *; clipboard-write';
```

Comment:
I remember that browser console was complaining about `publickey-credentials-create *;` is invalid. You sure this is where it should be at?

Reply:
Good catch, removed from now - this was added to the spec but hasn't been implemented by most modern browsers yet https://github.com/w3c/webauthn/issues/1656#issuecomment-1528147812